docs/tutorials/dataassociation/KDTree_Tutorial.py

```diff
@@ -0,0 +1,403 @@
+#!/usr/bin/env python
+# coding: utf-8
+"""
+=================================================
+11 - :math:`k`-d trees and TPR-trees
+=================================================
+"""
+
+# %%
+# :math:`k`-d trees and TPR-trees are both data structures which can be used to index objects in multi-dimensional
+# space.
+# 
+# - A :math:`k`-d tree is a binary search tree that operates in :math:`k` dimensions where each node represents a
+#   :math:`k`-dimensional point, where :math:`k` represents any number of dimensions. A :math:`k`-d tree is useful for
+#   indexing the positions of **static** objects.
+# - A TPR-tree (a time parameterised range tree) is a binary search tree which stores information about the object's
+#   position, direction, and velocity in substructures called Velocity Bounding Rectangles (VBRs). This information can
+#   be used to compute both current and predicted positions of the object. A TPR-tree is therefore useful for indexing
+#   the positions of **moving** objects.
+# 
+# NB: a :math:`k`-d tree can also be used to index the positions of moving objects, but it will only show a snapshot of
+# positions at a given timestep. The :math:`k`-d tree must be reconstructed at each time step, but the TPR tree only
+# needs to be reconstructed if there are any changes to an object's direction and velocity.
+# 
+# Both trees can increase the efficiency of searches for information relating to a given object when compared with
+# linear searches.
+
+
+
+# %%
+# Using :math:`k`-d trees for target tracking
+# -------------------------------------------
+# So far across the other tutorials, we have seen linear Nearest Neighbour and Probabilistic Data Association search
+# algorithms applied to target tracking to identify which detections should be associated with our target track(s).
+# 
+# During training of a basic Nearest Neighbour algorithm with a given training set of detections from a sensor, the
```

Comment:
I was slightly confused by the term "training" used in this paragraph. I think it'll be better to re-structure this paragraph by giving a simple, general overview of Nearest Neighbour and then a hyperlink to tutorial 5 showing how it's used in target-tracking / Stone Soup.